tests/unit_tests/simtel/test_simtel_config_writer.py

```diff
@@ -278,3 +278,33 @@ def test_write_random_seeds_file(simtel_config_writer, tmp_test_directory):
             if line[0] == "#":
                 continue
             assert line.strip().isdigit()
+
+    sim_telarray_seeds = {
+        "seed": 12345,
+        "seed_file_name": "sim_telarray_instrument_seeds.txt",
+        "random_instances": 1025,
+    }
+    with pytest.raises(
+        ValueError, match="Number of random instances of instrument must be less than 1024"
+    ):
+        simtel_config_writer._write_random_seeds_file(sim_telarray_seeds, config_file_directory)
+
+
+def test_sim_telarray_random_seeds():
```

Comment:
After fixing the case where zero is allowed as seed (if indeed I am right), a test for that should be added (though it will be meaningless because the chances of getting zero are too small).

Reply:
Well, what you are essentially testing in this function is that `rng.integers(low=1, high=max_int32 + 1, size=number, dtype=np.int32))` returns a number which is `np.int32`, so your comment is a bit funny. 

If we look beyond the internal implementation though, testing that `sim_telarray_random_seeds` doesn't return any zeros is not crazy and would be just adding a line like `assert all(s >=1 for s in seeds)`.